src/undocommand.js

```diff
@@ -54,13 +59,26 @@ export default class UndoCommand extends Command {
 	 * Executes the command: reverts a {@link engine.treeModel.Batch batch} added to the command's stack,
 	 * applies it on the document and removes the batch from the stack.
 	 *
-	 * @private
+	 * @protected
 	 * @fires undo.undoCommand#event:revert
-	 * @param {Number} [batchIndex] If set, batch under the given index on the stack will be reverted and removed.
-	 * If not set, or invalid, the last added batch will be reverted and removed.
+	 * @param {engine.treeModel.Batch} [batch] If set, batch that should be undone. If that batch is not on undo stack,
+	 * the command execution won't do anything. If not set, the last added batch will be undone.
 	 */
-	_doExecute( batchIndex ) {
-		batchIndex = this._batchStack[ batchIndex ] ? batchIndex : this._batchStack.length - 1;
+	_doExecute( batch = null ) {
+		let batchIndex;
+
+		// If batch is not given, set `batchIndex` to the last index in command stack.
+		// If it is given, find it on the stack.
+		if ( !batch ) {
+			batchIndex = this._batchStack.length - 1;
+		} else {
+			for ( let i = 0; i < this._batchStack.length; i++ ) {
```

Comment:
You could've used: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find. I'll fix it.